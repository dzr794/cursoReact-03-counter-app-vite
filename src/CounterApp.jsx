import PropTypes from 'prop-types'
import { useState } from 'react'

export const CounterApp = ( {value} ) => {

  const [ counter, setCounter ] = useState( value );

  const handleSubtract = () => {
    setCounter( counter - 1 );
  }

  const handleAdd = () => {
    setCounter( counter + 1 );
    // setCounter( (c) => c + 1 );
  }

  const handleReset = () => {
    setCounter( value );
  }

  return (
    <>
    <h1>CounterApp</h1>
    <h2 data-testid="counter-value" > { counter } </h2>
    <button data-testid="substract-1" onClick={handleSubtract}>-1</button>
    <button data-testid="add-1" onClick={handleAdd}>+1</button>
    <button data-testid="reset" onClick={handleReset}>Reset</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

CounterApp.defaultProps = {
  value: 0,
}