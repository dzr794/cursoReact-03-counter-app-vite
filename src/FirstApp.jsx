import PropTypes from 'prop-types'

export const FirstApp = ( { title, subtitle, name } ) => {
  return (
    <>
    <h1> {title} </h1>
    <p> {subtitle} </p>
    <p> {name} </p>
    </>
  )
}

FirstApp.propTypes = {
  name: PropTypes.string,
  subtitle: PropTypes.string,
  title: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name: "David",
  subtitle: "No hay subtitulo",
  title: "No hay t√≠tulo",
}