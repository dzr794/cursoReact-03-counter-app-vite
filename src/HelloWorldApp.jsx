// export function App() {
//   return (<h1>Holiwa !!!</h1>);
// }

// * usando el snipet "rafc"

const myFunction = (a,b) => a+b;

export const HelloWorldApp = () => {
  return (
    <h1>{myFunction(1,2)}</h1>
  )
}
