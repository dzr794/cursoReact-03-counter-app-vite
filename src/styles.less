@font-color: #f1f1f1;

html{
  body{
    background-color: #212121;
    color: @font-color;
    font-family: Helvetica, Arial, sans-serif;
    padding: 30px;
  }
}