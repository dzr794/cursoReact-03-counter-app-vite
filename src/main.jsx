import React from "react";
import ReactDOM from "react-dom/client"
// import { HelloWorldApp } from "./HelloWorldApp";
// import { FirstApp } from "./FirstApp";
import { CounterApp } from "./CounterApp";

import './styles.less';

ReactDOM.createRoot( document.getElementById('root') ).render(
  <React.StrictMode>
    {/* <FirstApp title="Holi"/> */}
    <CounterApp value={20} />
  </React.StrictMode>
);