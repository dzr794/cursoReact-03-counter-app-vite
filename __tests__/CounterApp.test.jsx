import {fireEvent, render, screen} from '@testing-library/react'
import { CounterApp } from '../src/CounterApp';

describe('pruebas en <CounterApp />', () => { 
  
  const initVal = 100;

  test('should match with the snapshot', () => {
    const { container } = render( 
      <CounterApp value={initVal} /> 
    );

    expect(container).toMatchSnapshot();
  });

  test('Init value must be 100', () => {
    render( 
      <CounterApp value={initVal} /> 
    );

    expect(screen.getByTestId("counter-value").innerHTML).toContain('100');
  });

  test('Should add 1 to counter value on Click', () => {
    render( 
      <CounterApp value={initVal} /> 
    );

    fireEvent.click( screen.getByTestId('add-1') );
    // screen.debug();
    
    expect( parseInt(screen.getByTestId('counter-value').innerHTML) ).toBe(initVal + 1)
  });

  test('Should substract 1 to counter value on Click', () => {
    render( 
      <CounterApp value={initVal} /> 
    );

    fireEvent.click( screen.getByTestId('substract-1') );
    // screen.debug();
    
    expect( parseInt(screen.getByTestId('counter-value').innerHTML) ).toBe(initVal - 1)
  });

  test('Should reset the counter value on Click', () => {
    render( 
      <CounterApp value={initVal} /> 
    );

    fireEvent.click( screen.getByTestId('add-1') );
    fireEvent.click( screen.getByTestId('add-1') );
    fireEvent.click( screen.getByTestId('add-1') );
    fireEvent.click( screen.getByTestId('reset') );
    // screen.debug();
    
    expect( parseInt(screen.getByTestId('counter-value').innerHTML) ).toBe(initVal)
  });

});