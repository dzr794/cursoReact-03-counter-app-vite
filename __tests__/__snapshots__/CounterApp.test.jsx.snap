// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <CounterApp /> should match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2
    data-testid="counter-value"
  >
     
    100
     
  </h2>
  <button
    data-testid="substract-1"
  >
    -1
  </button>
  <button
    data-testid="add-1"
  >
    +1
  </button>
  <button
    data-testid="reset"
  >
    Reset
  </button>
</div>
`;
